checker.cpp

```diff
@@ -1,42 +1,56 @@
 #include <iostream>
 #include <assert.h>
 #include <unistd.h>
+
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
-    }
+void flush()
+{
+	for (int i = 0; i < 6; i++)
+	{
+		cout << "\r* " << flush;
+		sleep(1);
+		cout << "\r *" << flush;
+		sleep(1);
+	}
+}
+
+int TemperatureCheck(float temperature)
+{
+	if (temperature > 102 || temperature < 95) 
+	{
+		cout << "Temperature critical!\n";
+		flush();
+		return 0;
+	}
+	return 1;
+}
+
+int PulseCheck(float pulseRate)
+{
+	if (pulseRate < 60 || pulseRate > 100)
```

Comment:
Consider moving the range-check to a common place, to make it easy to implement warning levels